src/iterative_telemetry/__init__.py

```diff
@@ -187,39 +187,60 @@ def _system_info():
     raise NotImplementedError
 
 
+def _generate_id():
+    """A randomly generated ID string"""
+    return str(uuid.uuid4())  # TODO: CI env-based ID
+
+
+def _read_user_id(config_file: Path):
+    try:
+        with config_file.open(encoding="utf8") as fobj:
+            return json.load(fobj)["user_id"]
+    except (FileNotFoundError, ValueError, KeyError):
+        pass
+    return None
+
+
+def _read_user_id_locked(config_file: Path):
+    lockfile = str(config_file.with_suffix(".lock"))
+    if config_file.parent.is_dir():
+        with FileLock(lockfile, timeout=5):
```

Comment:
Timeout exception from here will fall through and result in an error that will say that it failed to acquire new lockfile when in fact this is about an old lockfile.

Reply:
indeed, the "fall through and result in an error (though really a debug message) that will say that it failed to acquire" is intentional, but the "new lockfile when in fact this is about an old lockfile" is indeed misleading. Fixed now :)